### PR TITLE
Add Meta Pixel app

### LINKED ISSUE #3563: Facebook Pixel implementation
### Is your proposal related to a problem?

A client of mine askes if there is the posibility of inserting facebook pixel to the different events.

### Describe the solution you'd like

If we can have a general Pixel insert in the main setting and then for each event to be able to create the event tag for facebook.

### Describe alternatives you've considered

I am not good at coding, I looked through the code, but did not know where to insert it. Facebook makes it quite easy to insert it, but I did not know where on this end.

### Additional context

Thanks for your help.

### PR DESCRIPTION
**Pull request by regexyl**
## What does this PR do?
This PR adds a Meta Pixel app which records a page view event under the custom event 'CalcomView', corresponding to the organizer's Pixel ID.

Fixes #3563
/claim #3563

**Loom Video**
https://www.loom.com/share/373dfbbaa00d40d98c509d98b3f9b144

**EventType --> Apps screenshot**
image

**Event registered in Facebook Events Manager**
image
image

**Environment**: Staging(main branch) / Production

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [x] Install the Meta Pixel app
- [x] Add the app to an event, insert the Meta Pixel ID
- [x] Ensure Meta Pixel script is visible in the booking page
- [x] Check if the CalcomView event is recorded in the Meta/Facebook Events Manager

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/app-store/metapixel/config.json
Comment (hariombalhara) [inline packages/app-store/metapixel/config.json:13]:
```suggestion
  "description": "Add Meta Pixel to your bookings page to measure, optimize and build audiences for your ad campaigns.",
```

File: packages/app-store/metapixel/config.json
Comment (hariombalhara) [inline packages/app-store/metapixel/config.json:17]: This is deprecated and unused.
```suggestion
```

File: packages/app-store/metapixel/config.json
Comment (hariombalhara) [inline packages/app-store/metapixel/config.json:18]:
```suggestion
```
dirName isn't required

File: yarn.lock
Comment (hariombalhara) [inline yarn.lock:8459]: I regenerated yarn.lock after pulling latest submodules and it seems to update these cc @emrysal @zomars

**Review by hariombalhara — APPROVED** [review]
Looks good to me. Thank you for the contribution @regexyl 🙏
I have made some changes that ensure that dead/deprecated code isn't added in the new app. Those  improvements would land in App Store CLI and templates

Tests that I ran
- App can be enabled for installation from Admin->Apps
- App can be installed from Event Type -> Apps and after enabling it for the Event Type and providing Pixel ID, the booker page sends a request to` fbevents.js`

One thing that's missing in loom as well as in screenshots is showing that `CalcomView` event is there in some Meta dashboard. I would appreciate a screenshot for that.

Also, a screenshot for showing the App in EventType->Apps as enabled with a sample Pixel ID could also be useful.

cc @PeerRich for final approval

**Comment by regexyl** [comment]
Thanks a lot @hariombalhara!

I've tried testing it locally but AFAIK I don't think Facebook Events Manager allows for the recording of localhost events:
<p float="left">
image
image
</p>

I tried copying the same injected script in my own domain for testing (+ changed the event name to `Test24AprView`) and it was registered. It's not indicative of the actual thing, but would this be sufficient?
image

I've just updated the PR description with the EventType --> Apps screenshot too.

**Comment by regexyl** [comment]
That works great, thanks - I've modified by hosts file and received a CalcomView from my local environment after landing on the booking page:

image

**Comment by hariombalhara** [comment]
Awesome !! Let's add that screenshot to the DESCRIPTION.md and wait for final review from @PeerRich

**Review by PeerRich — APPROVED** [review]

**Comment by danielehrhardt** [comment]
I cannot see this in the App Store: https://app.cal.com/apps
What do i miss?

**Comment by hariombalhara** [comment]
Could you check again, it should be accessible now

**Comment by growth-wizard-digital** [comment]
Does this just push the 'calcomview' event, or does it also register bookings/date selections?

**Comment by regexyl** [comment]
@growth-wizard-digital Just the CalcomView event for now!

**Pull request merged**
